proposals/0002-root-signature-in-clang.md

```diff
@@ -936,6 +936,199 @@ Operands:
+carried forward to Clang and their expected behaviour.
+
+(Design notes and unplanned options will be removed after being resolved)
+
+#### Option `-force-rootsig-ver`
+
+When compiling a shader with a root signature, this option overrides the root
+signature version used, where the default is `rootsig_1_1`.
+
+Usage:
+
+```
+  -force-rootsig-ver rootsig_1_0
+  -force-rootsig-ver rootsig_1_1
+```
+
+Behaviour:
+
+ - Validation logic that is specific to the version will be followed, as
+described above
+ - Serialized root signature in DXIL container will be compliant with specified
+version
+
+Design Notes: Required for backwards compatibility. Implemented in Clang.
+
+#### Option `-rootsig-define`:
+
+Sets the root signature attribute for the entry function to be the root
+signature defined by the given macro expansion.
+
+Usage:
+
+Given a defined macro either provided in the source file
+(`#define RS "CBV(b0)"`), or, as a command line define
+(`-D RS="CBV(b0)"`)
+
+```
+  -rootsig-define RS
+```
+
+Behaviour:
+
+ - If the entry function does not have a root signature attribute, it will use
+the one defined by the macro expansion
+
+ - If the entry function has a root signature attribute, it will overwrite to
+use the one defined by the macro expansion
+
+Design Notes: Commonly used. Planned in Clang.
+
+#### Target Root Signature Version
+
+Compiles the "entry" root signature, specified by the given macro expansion, to
+a DXIL Container with just the (version specific) RTS0 part.
+
+Usage:
+
+Given a defined macro in the source file (`#define RS "CBV(b0)"`)
+
+```
+  -T <root signature version> -E <entry root signature>`
+  -T rootsig_1_0 -E RS /Fo RS.bin
+  -T rootsig_1_1 -E RS /Fo RS.bin
+```
+
+Behaviour:
+
+ - Parse and perform syntactic validations of "entry" root signature
+ - Perform the 'basic' sub-set of validations that are available without access
+to resource bindings
+ - Produces a DXIL container with only the serialized "entry" root signature
+as the only part
+
+ - Note: It is not possible to use `-rootsig-define` to overwrite which root
+signature will be used as the "entry" root signature.
+
+Design Notes: Commonly used. Planned in Clang.
+
+#### Option `-Qstrip-rootsignature`
+
+Omits the root signature part (RTS0) from the produced DXIL Container.
+
+Usage:
+
+```
+  -Qstrip-rootsignature
+```
+
+Behaviour:
+
+ - Parse and perform syntactic validations of "entry" root signature
+ - Perform the 'basic' sub-set of validations that are available without access
+to resource bindings
+ - Produces the DXIL container with the RTS0 omitted
+
+Design Notes: Usable within .hlsl source files. Planned in Clang unless there is
+a signal of it being unused.
+
+Design Notes: Plan to match DXC behaviour directly such that we still do
+the parsing and basic validations of the root signataure.
+
+#### Option `-setrootsignature`
+
+Overwrite the root signature part (RTS0) with the RTS0 of provided DXIL
+Container.
```

Comment:
This can be used to override the root signature part used either in the resulting DXIL container when compiling a shader, or by adding or replacing the root signature part of an already compiled shader in a DXIL Container.

```suggestion
Override the root signature part (RTS0) in the DXIL Container produced either
when compiling a shader, or in a container loaded using the `-dumpbin` option.
```